src/cascadia/TerminalSettingsEditor/ActionsViewModel.cpp

```diff
@@ -18,383 +24,1380 @@ using namespace winrt::Windows::UI::Xaml::Data;
+                        }
+
+                        if (flagWrapper.IsSet())
+                        {
+                            nullEntry.IsSet(false);
+                            WI_SetAllFlags(localUnboxed, flagValue);
+                        }
+                        else
+                        {
+                            WI_ClearAllFlags(localUnboxed, flagValue);
+                        }
+                        Value(winrt::box_value(winrt::Windows::Foundation::IReference<EnumType>(localUnboxed)));
                     }
-                    else
+                });
+                flagList.emplace_back(entry);
+            }
+        }
+        std::sort(flagList.begin(), flagList.end(), winrt::Microsoft::Terminal::Settings::Editor::implementation::FlagEntryReverseComparator<EnumType>());
+        // nullEntry handler that toggles/clears other flags
+        nullEntry.PropertyChanged([this](const winrt::Windows::Foundation::IInspectable& sender,
+                                         const winrt::Windows::UI::Xaml::Data::PropertyChangedEventArgs& args) {
+            const auto itemProperty = args.PropertyName();
+            if (itemProperty == L"IsSet")
+            {
+                auto flagWrapper = sender.as<winrt::Microsoft::Terminal::Settings::Editor::FlagEntry>();
+                if (flagWrapper.IsSet())
+                {
+                    for (const auto& flagEntry : _FlagList)
                     {
-                        // Exit edit mode for all other containers
-                        get_self<KeyBindingViewModel>(kbdVM)->DisableEditMode();
+                        if (flagEntry.FlagName() != RS_(L"Actions_NullEnumValue"))
+                        {
+                            flagEntry.IsSet(false);
+                        }
                     }
+                    Value(winrt::box_value(winrt::Windows::Foundation::IReference<EnumType>(nullptr)));
+                }
+                else
+                {
+                    Value(winrt::box_value(winrt::Windows::Foundation::IReference<EnumType>(EnumType{ 0 })));
                 }
             }
-            else
+        });
+        flagList.emplace_back(nullEntry);
+        _FlagList = winrt::single_threaded_observable_vector<winrt::Microsoft::Terminal::Settings::Editor::FlagEntry>(std::move(flagList));
+    }
+
+    ActionArgsViewModel::ActionArgsViewModel(const Model::ActionAndArgs actionAndArgs) :
+        _actionAndArgs{ actionAndArgs }
+    {
+    }
+
+    void ActionArgsViewModel::Initialize()
+    {
+        const auto shortcutArgs = _actionAndArgs.Args().as<Model::IActionArgsDescriptorAccess>();
+        if (shortcutArgs)
+        {
+            const auto shortcutArgsDescriptors = shortcutArgs.GetArgDescriptors();
+            std::vector<Editor::ArgWrapper> argValues;
+            uint32_t i = 0;
+            for (const auto argDescription : shortcutArgsDescriptors)

```

Comment:
make sure you use `&` for this, otherwise you're going to copy the entire descriptor.

You can also probably destructure it with...

```
const auto& [argName, argType, argRequired, argTypeHint] = argDescription;
```

though it raises the question: why not add a constructor to `ArgWrapper` which just takes a `const ArgDescription&` directly?